vortex-array/src/array/varbin/canonical.rs

```diff
@@ -23,11 +23,29 @@ impl IntoCanonical for VarBinArray {
         VarBinViewArray::try_from(ArrayData::from_arrow(array, nullable)).map(Canonical::VarBinView)
     }
 
-    fn into_arrow(self) -> VortexResult<ArrayRef> {
+    fn into_arrow(self) -> VortexResult<ArrayRef>
+    where
+        Self: Sized,
+    {
         // Specialized implementation of `into_arrow` for VarBin since it has a direct
         // Arrow representation.
         varbin_to_arrow(&self)
     }
+
+    fn into_arrow_with_data_type(self, data_type: &DataType) -> VortexResult<ArrayRef> {
+        let array_ref = self.into_arrow()?;
+
+        // Note, arrow::cast clones the array, so don't use it if unnecessary.
+        Ok(match data_type {
+            DataType::Binary | DataType::LargeBinary | DataType::Utf8 | DataType::LargeUtf8 => {
+                array_ref
```

Comment:
This doesn't make much sense to me. You switch on the requested data type, then you return whatever into_arrow returns without checking the data_type actually matches?

Shouldn't it be more like `if array_ref.data_type() != data_type { array_ref = cast(...) }`

Reply:
agreed